File: app/src/main/java/com/timotiusoktorio/pencake/ui/BaseActivity.java
package com.timotiusoktorio.pencake.ui;

import android.os.Bundle;
import android.support.annotation.Nullable;
import android.support.v7.app.AppCompatActivity;

import com.timotiusoktorio.pencake.MyApplication;
import com.timotiusoktorio.pencake.di.ActivityComponent;
import com.timotiusoktorio.pencake.di.ActivityModule;
import com.timotiusoktorio.pencake.di.DaggerActivityComponent;

public abstract class BaseActivity extends AppCompatActivity {

    private ActivityComponent component;

    @Override
    protected void onCreate(@Nullable Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        component = DaggerActivityComponent.builder()
                .appComponent(MyApplication.get(this).getComponent())
                .activityModule(new ActivityModule(this))
                .build();
    }

    public ActivityComponent getComponent() {
        return component;
    }
}

File: app/src/main/java/com/timotiusoktorio/pencake/ui/BaseActivity.kt
package com.timotiusoktorio.pencake.ui

import android.os.Bundle
import android.support.v7.app.AppCompatActivity

import com.timotiusoktorio.pencake.MyApplication
import com.timotiusoktorio.pencake.di.ActivityComponent
import com.timotiusoktorio.pencake.di.ActivityModule
import com.timotiusoktorio.pencake.di.DaggerActivityComponent

abstract class BaseActivity : AppCompatActivity() {

    lateinit var component: ActivityComponent

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        component = DaggerActivityComponent.builder()
                .appComponent(MyApplication[this].component)
                .activityModule(ActivityModule(this))
                .build()
    }
}